crates/notary/server/src/main.rs

```diff
@@ -1,25 +1,41 @@
 use eyre::{eyre, Result};
 use structopt::StructOpt;
-use tracing::debug;
-
+use tracing::{info, debug, error};
 use notary_server::{
-    init_tracing, parse_config_file, run_server, CliFields, NotaryServerError,
-    NotaryServerProperties,
+    init_tracing, run_server, CliFields, NotaryServerError,
+    Settings
 };
 
 #[tokio::main]
 async fn main() -> Result<(), NotaryServerError> {
-    // Load command line arguments which contains the config file location
+    // Load command line arguments
     let cli_fields: CliFields = CliFields::from_args();
-    let config: NotaryServerProperties = parse_config_file(&cli_fields.config_file)?;
+
+    let settings = match Settings::new(&cli_fields) {
+        Ok(s) => s,
+        Err(e) => {
+            error!("Failed to load settings: {:?}. Check that all required fields are provided in the configuration.", e);
```

Comment:
the previous code was better and more succinct, curious why did you change into this?

```rust
let settings = Settings::new(&cli_fields)
        .map_err(|err| eyre!("Failed to load settings: {}", err))?;
```